src/components/application_manager/src/application_data_impl.cc

```diff
@@ -582,6 +582,121 @@ void DynamicApplicationDataImpl::set_display_layout(const std::string& layout) {
   window_templates_[DEFAULT_WINDOW] = template_config;
 }
 
+/**
+ * @brief RemoveUnsupportedLayouts removes unsupported layouts from
+ * configurable_keys array, if present
+ * @param keyboard_capabilities Modified keyboard capabilities
+ * @param supported_layouts Set with all supported layouts, generated from
+ * supported_keyboard_layouts array
+ * @param customizable_layouts Set with all supported layouts, which keys may be
+ * customized (if not, number of keys for such layout will be set to zero by
+ * default)
+ */
+void RemoveUnsupportedLayouts(
+    smart_objects::SmartObject& keyboard_capabilities,
+    const std::set<mobile_apis::KeyboardLayout::eType>& supported_layouts,
+    std::set<mobile_apis::KeyboardLayout::eType>& customizable_layouts) {
+  SDL_LOG_AUTO_TRACE();
+
+  if (!keyboard_capabilities.keyExists(hmi_response::configurable_keys)) {
+    SDL_LOG_WARN(
+        "No configurable keys param in received keyboard capabilities");
+    return;
+  }
+
+  auto configurable_keyboards =
+      keyboard_capabilities[hmi_response::configurable_keys].asArray();
+  if (!configurable_keyboards) {
+    return;
+  }
+
+  for (auto keyboard = (*configurable_keyboards).begin();
+       keyboard != (*configurable_keyboards).end();) {
+    auto found_layout =
+        supported_layouts.find(static_cast<mobile_apis::KeyboardLayout::eType>(
+            (*keyboard)[hmi_request::keyboard_layout].asInt()));
+    if (supported_layouts.end() == found_layout) {
+      (*configurable_keyboards).erase(keyboard);
+    } else {
+      customizable_layouts.insert(
```

Comment:
@ychernysheva I think would be better to split this function into two separate functions: one for `configurable_keyboards` modifying and one for `customizable_layouts` modifying

Reply:
Done in a9c3f0b